zerver/lib/users.py

```diff
@@ -511,6 +515,45 @@ def check_user_can_access_all_users(acting_user: Optional[UserProfile]) -> bool:
     return False
 
 
+def check_can_access_user(
+    target_user: UserProfile, user_profile: Optional[UserProfile] = None
+) -> bool:
+    if check_user_can_access_all_users(user_profile) or not user_access_restricted_in_realm(
+        target_user
+    ):
+        return True
+
+    assert user_profile is not None
```

Comment:
Presumably this just reflects that we haven't tried to make this work for spectators yet, right?

Reply:
We allow spectators access to all users so `if check_user_can_access_all_users` would return true.

Reply:
Yeah. I can imagine a world post client capability where we start trying to use this code path with spectators for performance reasons, but we can certainly leave that for later.